MIP/mip-58/README.md

```diff
@@ -0,0 +1,208 @@
+   - The current design requires users to have funds on the destination chain for finalization. This increases friction and discourages adoption.
+   - Sponsored transactions are essential to the current implementation but remain unimplemented, leaving users stranded without funds.
+4. **Security Risks**:
+   - Relayer keys are a critical point of failure. This assumption cannot be avoided.
+   - Refund logic introduces vulnerabilities where attackers could exploit the rate limit and relayer downtime.
+   - For instance, if the relayer fails to finalize on the initiator, a malicious actor could:
+     - Take over the refund keys.
+     - Exploit both directions of the bridge (e.g., ETH→MOVE and MOVE→ETH) repeatedly, draining funds.
+5. **Maintenance Burden**:
+   - The current audit is outdated and does not reflect the significant changes made since.
+   - The current design requires a complete UI/UX overhaul, adding complexity and delay.
+6. **Unnecessary Complexity**:
+   - HTLC-based bridges are largely abandoned in favor of simpler, more effective designs.
+   - Examples like the [Consensys HTLC](https://github.com/Consensys/htlc-bridge) bridge demonstrate the [pitfalls of such approaches](https://entethalliance.org/crosschain-bridges-overview-where-we-are-now/).
+7. **Infrastructure Simplification**:
+   - Infrastructure is still incomplete for the current HTLC bridge design and we are struggling to find sound solutions that balance UI/UX and security. For example, [this issue](https://github.com/movementlabsxyz/movement/issues/838) regarding UX vs number of confirmations to require on the Ethereum side, is still under debate.
+   - Because of the over-engineered design, infrastructure is prone to error and we might end up being damaged by the amount of infrastructure we have not built yet and has to be built for the relayer to fully function.
+   - We could strip down the Relayer code and achieve a final design much more quickly.
+
+The proposed lock/mint bridge design mitigates these issues, creating a safer, faster, and user-friendly bridge while maintaining operational reliability.
+
+## Specification
+
+The lock/mint bridge design focuses on minimizing complexity and maximizing security:
+
+### Protocol Description
+
+The bridge is a simpler version of the current implementation, which relies on the assumption that the Relayer Keys are not to be compromised.
+
+L1 -> L2
+
+1. User initiates a bridge transfer on L1. Contract stores a mapping of the user bridgeTransferIds for easy access. It transfers from the user MOVE amount to the contract. Transaction emits originator, recipient, amount and nonce.
+2. Relayer awaits for finalization of the transaction on L1.
+3. Relayer completes the bridge on L2.
+4. Completion transaction verifies that the transaction is truthful by comparing the provided bridgeTransferId hash and the emitted values of originator, recipient, amount and nonce. Finally, it mints MOVE to the recipient address.
```

Comment:
The method for checking the parameters needs to be specified. Where are the code samples we put in? Those should be restored here and the L2 -> L1 steps (for each, include initiate function code in step 1 and complete function code in step 4). Right now these steps are unacceptably vague; it doesn't specify how the comparison happens. Linking to a couple GitHub repositories later in the document is not sufficient. The code should be here in the protocol description, or at least there should be an exact description of how the comparison happens.